### PR TITLE
[device][platform] add platform as5812-54x, accton.

### PR DESCRIPTION
**Pull request by roylee123**
Add support for a platform, as5812-54x:
CPU: Intel Atom® Processor C2538
MAC: Broadcom Trident2+(BCM56864), 720Gbs BW.
BMC: None

**- What I did**
Add device and platform files for as5812-54x.
.
**- How I did it**
Add as SONiC Porting Guide.

**- How to verify it**
decode-syseeprom
sensors
psuutil
sfputil
thermal monitor
(port LEDs may not working)

**- Description for the changelog**
As5812-54x shared most of design of as5712-54x, but the MAC chip.
But it also find QSFP's low-power mode is missed at last change of https://github.com/Azure/sonic-buildimage/pull/2110.  Add it back and tested.

### CODE REVIEW

File: device/accton/x86_64-accton_as5812_54x-r0/Accton-AS5812-54X/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# name         lanes              alias         index
+Ethernet0      13                 tenGigE0      0
```
Comment (lguohan) [inline device/accton/x86_64-accton_as5812_54x-r0/Accton-AS5812-54X/port_config.ini]: why does the index start from 0? are your front panel port numbering start from 0?

**Comment by lguohan** [comment]
can you remove changes to mellanox and p4?

File: device/accton/x86_64-accton_as5812_54x-r0/Accton-AS5812-54X/port_config.ini (hunk 1 above)
Comment (roylee123) [reply device/accton/x86_64-accton_as5812_54x-r0/Accton-AS5812-54X/port_config.ini]: No, the port number on the panel starts from 1. 
Most port_config.ini of accton devices starts from Ethernet0,  I just follow the way we used to be.

Comment (lguohan) [reply device/accton/x86_64-accton_as5812_54x-r0/Accton-AS5812-54X/port_config.ini]: check here. https://github.com/Azure/SONiC/wiki/Porting-Guide

index: Physical index of port (optional - needed if ports do not begin with index zero or increment by one), When this column is missing the index starts from 0 and increase by 1 for each row. It is recommend to explicitly set this index column and make the index number match the labeling of the front panel ports. AKA: if the front panel ports are starting from 1, the index should start from 1. If the front panel ports start from 0, the index should start form 0. The break out ports from the same physical port should have same index.

**Comment by lguohan** [comment]
can you resolve conflict?

**Pull request merged**